src/production/transformers/__tests__/Phase_1_Import.test.ts

```diff
@@ -0,0 +1,299 @@
+import { testProductionWarmup } from '../testUtils';
+
+function test(code: string) {
+  return testProductionWarmup({
+    code,
+    transformers: [Phase_1_ImportLink()],
+    moduleProps: {
+      moduleSourceRefs: {
+        './foo': {
+          some: 'value'
+        }
+      }
+    }
+  });
+}
+
+describe('Phase 1 - Imports test', () => {
+  it(`sideEffectImport import './foo'`, () => {
+    const { tree } = test(`
+      import './foo';
+    `);
+    const refs = tree.importReferences.references;
+
+    expect(refs).toHaveLength(1);
+    expect(refs[0].source).toEqual('./foo');
+    expect(refs[0].type).toEqual(ImportReferenceType.SIDE_EFFECT_IMPORT);
+  });
+
+  it(`sideEffectImport import './foo' should be removed`, () => {
+    const { tree } = test(`
+      import './foo';
+    `);
+    const refs = tree.importReferences.references;
+
+    expect(refs).toHaveLength(1);
+    refs[0].remove();
+    expect(refs[0].removed).toBe(true);
+    refs.splice(0, 1);
+    expect(refs).toHaveLength(0);
+  });
+
+  it(`regularImport import foo from './foo'`, () => {
+    const { tree } = test(`
+      import foo from './foo';
+    `);
+    const refs = tree.importReferences.references;
+
+    expect(refs).toHaveLength(1);
+    expect(refs[0].source === './foo');
+    expect(refs[0].specifiers).toHaveLength(1);
+    expect(refs[0].specifiers[0].local === 'foo');
+    expect(refs[0].specifiers[0].name === 'default');
+  });
+
+
+  it(`regularImport import { foo, bar as baz } from './foo'`, () => {
+    const { tree } = test(`
+      import { foo, bar as baz } from './foo';
+    `);
+    const refs = tree.importReferences.references;
+
+    expect(refs).toHaveLength(1);
+    expect(refs[0].source === './foo');
+    expect(refs[0].specifiers).toHaveLength(2);
+    expect(refs[0].specifiers[0].local === 'foo');
+    expect(refs[0].specifiers[0].name === 'foo');
+    expect(refs[0].specifiers[1].local === 'baz');
+    expect(refs[0].specifiers[1].name === 'bar');
+  });
+
+  it(`regularImport import foo, { bar } from './foo'`, () => {
+    const { tree } = test(`
+      import foo, { bar } from './foo';
+    `);
+    const refs = tree.importReferences.references;
+
+    expect(refs).toHaveLength(1);
+    expect(refs[0].source === './foo');
+    expect(refs[0].specifiers).toHaveLength(2);
+    expect(refs[0].specifiers[0].local === 'foo');
+    expect(refs[0].specifiers[0].name === 'default');
+    expect(refs[0].specifiers[1].local === 'bar');
+    expect(refs[0].specifiers[1].name === 'bar');
+  });
+
+  it(`regularImport import * as bar from './foo'`, () => {
+    const { tree } = test(`
+      import * as bar from './foo';
+    `);
+    const refs = tree.importReferences.references;
+
+    expect(refs).toHaveLength(1);
+    expect(refs[0].source === './foo');
+    expect(refs[0].specifiers).toHaveLength(1);
+    expect(refs[0].specifiers[0].local === 'bar');
+    expect(refs[0].specifiers[0].name === 'default');
```

Comment:
should be `undefined`

Reply:
source code updated. name is now undefined